web_api.md

```diff
@@ -0,0 +1,261 @@
+
+<figure>
+  <img src="images/web-api-distributed-queue.jpg" alt="Distributed architecture with the Web API"/>
+  <figcaption align="center">
+    <b>Fig. 1:</b> OCR-D System Architecture
+  </figcaption>
+</figure>
+
+### Description
+
+As shown in Fig. 1, each section in the [Web API specification](#the-specification) is implemented by different servers,
+which are Discovery Server, Processing Broker, Workflow Server, and Workspace Server respectively. Although each server
+in the figure is deployed on its own machine, it is completely up to the implementors to decide which machines run which
+servers. However, having each processor run on its own machine reduces the risk of version and resource conflicts.
+Furthermore, the machine can be customized to best fit the processor's hardware requirements and throughput demand. For
+example, some processors need GPU computation, while others do not, or some need more CPU capacity while others need
+more memory. It is also easier to scale up the processors, or even apply Function-as-a-Service on some of them, which
+are not constantly used, to save resources.
+
+**Processing**: since the `Processing` section is provided by [OCR-D Core](https://github.com/OCR-D/core), implementors
+do not need to implement Processing Broker, Message Queue, and Processing Server. Once a request comes, the broker
+pre-processes it if necessary, and push it to an appropriate queue. A processing queue always has the same name as its
+respective processors. For example, `ocrd-olena-binarize` processors listen only to the queue
+named `ocrd-olena-binarize`. A Processing Server, which is
+an [OCR-D Processor](https://ocr-d.de/en/spec/glossary#ocr-d-processor) running as a worker, listens to the queue, pulls
+new jobs when available, processes them, and returns results. One do not call a Processing Server directly, but via a
+Processing Broker. Job statuses can be pushed back to the queue, depending on the [job configuration](#message-queue),
+so that other services get updates and act accordingly.
+
+**Database**: in this architecture, a database is required to store necessary information such as users requests, jobs
+statuses, workspaces, etc. We recommend to use [MongoDB](https://www.mongodb.com/) since it is used by Processing
+Servers, but other kinds of storage may work as well.
+
+**Network File System**: in order to avoid file transfer between different machines, it is highly recommended to have
+a [Network File System](https://en.wikipedia.org/wiki/Network_File_System) set up. With NFS, all Processing Servers
+(specifically processors) can work in a shared storage environment and access files as if they are local files. To get
+data into the NFS, one could use the `POST /workspace` endpoint to upload [OCRD-ZIP](https://ocr-d.de/en/spec/ocrd_zip)
+files. However, this approach is only appropriate for small data size. Usually, Workspace Server should be able to pull
+data from other storages.
```

Comment:
The detail implementation is up to the IP. In my mind, the endpoint accepts a URL (and credentials if necessary), then just download the zip file and extract it to create a workspace.

IPs can also implement that endpoint in a way that it accepts a URL to a METS file, download it, then download all files in METS to create a workspace.

I don't include this endpoint in the `openapi.yml` file, since I don't think it should be compulsory. IPs can decide if they want to implement it or not.

Reply:
But we do include the `/workspace` endpoint in the web api spec, don't we?

I think it would be beneficial for the IP reading this if you could add the information from [your comment](https://github.com/OCR-D/spec/pull/222#discussion_r1064891353) to the text here, so they get an idea of what is meant with "other storages" (better; "other storage solutions").